etc/profile-m-z/nodejs-common.profile

```diff
@@ -7,13 +7,16 @@ include nodejs-common.local
 # added by caller profile
 #include globals.local
 
-blacklist /tmp/.X11-unix
-blacklist ${RUNUSER}
-
 ignore noexec ${HOME}
 
+# Required to run `npx`
```

Comment:
Not sure I understand the reasoning here. We already have `noblacklist ${HOME}/.npm` in npm.profile. Is npx not working without repeating it here?

Reply:
I don't think `npm.profile` is ever read. `cat $(which npx)` shows it's a script starting with `#!/usr/bin/env node`.

Reply:
> I don't think npm.profile is ever read. cat $(which npx) shows it's a script starting with #!/usr/bin/env node.

Indeed it is. In fact, both /usr/bin/npm and /usr/bin/npx are symlinks to scripts with the node shebang (at least that's what I see on Arch Linux). Wouldn't that imply we need a `npx.profile` too, as well as a `node.profile`? Just asking, I'm not at all familiar with the Node.js stack.

Reply:
Regardless of the above remark, if we add `noblacklist ${HOME}/.npm` here, we should drop it from the `npm.profile`. Can you take care of that please? It's the only thing that needs to be done to merge this.

Reply:
`npx` being a node script, means firejail will read `node.profile`, not `npm.profile`.

> Wouldn't that imply we need a npx.profile too, as well as a node.profile

Correct. Or, we could remove `npm.profile`, and only create a `node.profile` which would be used for `node`, `npm` and `npx`. Though @rusty-snake pointed out in #4085 that there's only `nodejs-common.profile`, so I suppose there might be a good reason for that.

> if we add noblacklist ${HOME}/.npm here, we should drop it from the npm.profile. Can you take care of that please?

That would be moot if we get rid of [`npm.profile`](https://github.com/netblue30/firejail/blob/master/etc/profile-m-z/npm.profile). @aidalgol ?

Reply:
> That would be moot if we get rid of npm.profile. @aidalgol ?

It is not moot for a PR that adds it, which this does right?. Please let's not move ahead of ourselves and stick to the PR itself.

Reply:
> That would be moot if we get rid of [`npm.profile`](https://github.com/netblue30/firejail/blob/master/etc/profile-m-z/npm.profile). @aidalgol ?

Sorry, I am not familiar enough with firejail's general policies and strategies to comment.